var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -1122,6 +1122,10 @@ def configure_args(self):
         if "~romio" in spec:
             config_args.append("--disable-io-romio")
 
+        # Disable Lustre if requested
+        if "~lustre" in self.spec:
+            config_args.append("--without-lustre")
```

Comment:
Instead of the custom treatment, I would suggest moving the `lustre` and `singularity` variant handling from some 20 lines above this into the following consistent treatment with recommended `self.with_or_without`:
```suggestion
        for dep in ["lustre", "singularity"]:
            config_args.extend(self.with_or_without(dep, activation_value="prefix"))
```